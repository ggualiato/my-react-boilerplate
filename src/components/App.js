import React from "react";
import ReactIcon from "../images/react-icon.png";

const App = () => {
    const onClick = () => {
        alert("Hello World ");
    };

    return (
        <div className="app">
            <h3>Hello React- Basic project</h3>
            <img width="200" src={ReactIcon} />
            <button onClick={onClick}>Click here!</button>
        </div>
    );
};

export { App };
