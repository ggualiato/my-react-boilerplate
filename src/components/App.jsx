import React from "react";
import ReactIcon from "../images/react-icon.png";
import { sum } from "../utils/utils";

const App = () => {
    const onClick = () => {
        alert("Hello World " + sum(2, 3));
    };

    return (
        <div className="app">
            <h3>Hello React- Basic project</h3>
            <img width="200" src={ReactIcon} />
            <button onClick={onClick}>Click here!</button>
        </div>
    );
};

export { App };
