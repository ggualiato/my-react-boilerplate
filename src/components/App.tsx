import React, { FC, useEffect } from "react";
import ReactIcon from "../images/react-icon.png";
import Logo from "../images/logo.svg";
import { sum } from "../utils/utils";
import { Hello } from "./Hello";
import { sayHello } from "../utils/hello";
import { getTodo, getTodos } from "../services/todos";

const App: FC = () => {
    useEffect(() => {
        getTodos();
        getTodo(1);
    }, []);

    const onClick = () => {
        alert(sayHello(" World") + sum(2, 3));
    };

    return (
        <div className="app">
            <h3>Hello React- Basic project</h3>
            <div>
                <div className="image-wrapper">
                    PNG
                    <img width="200" src={ReactIcon} />
                </div>
                <div className="image-wrapper">
                    SVG
                    <a href="#" onClick={() => console.log("clicked")}>
                        <img width="200" src={Logo} />
                    </a>
                </div>
                <Hello onClick={onClick} text="Click here!" />
            </div>
        </div>
    );
};

export { App };
