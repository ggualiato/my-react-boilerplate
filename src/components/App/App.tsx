import React, { FC, useEffect } from "react";
import Logo from "../../images/logo.svg";
import ReactIcon from "../../images/react-icon.png";
import Hello from "../Hello/Hello";
import { getTodo, getTodos } from "./../../services/todos";
import { sayHello } from "./../../utils/hello";
import { sum } from "./../../utils/sum";
import { AppContainer, ImageWrapper } from "./styles";

const App: FC = () => {
    useEffect(() => {
        getTodos();
        getTodo(1);
    }, []);

    const onClick = () => {
        alert(sayHello(" World ") + sum(2, 3));
    };

    return (
        <AppContainer>
            <h3>Hello React- Basic project</h3>
            <div>
                <ImageWrapper>
                    PNG
                    <img width="200" src={ReactIcon} />
                </ImageWrapper>
                <ImageWrapper>
                    SVG
                    <a href="#" onClick={() => console.log("clicked")}>
                        <img width="200" src={Logo} />
                    </a>
                </ImageWrapper>
                <Hello onClick={onClick} text="Click here!" />
            </div>
        </AppContainer>
    );
};

export default App;
