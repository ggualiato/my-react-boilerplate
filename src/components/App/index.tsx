import React, { FC, useEffect } from "react";
import Logo from "../../images/logo.svg";
import ReactIcon from "../../images/react-icon.png";
import { getTodo, getTodos } from "../../services/todos";
import { sayHello, sum } from "../../utils";
import { Hello } from "../Hello";
import { AppContainer, Button, ImageWrapper } from "./styles";

export const App: FC = () => {
    useEffect(() => {
        getTodos();
        getTodo(1);
    }, []);

    const onClick = () => {
        alert(sayHello(" World ") + sum(2, 3));
    };

    return (
        <AppContainer>
            <h3>Hello React- Basic project</h3>
            <div>
                <ImageWrapper>
                    PNG
                    <img width="200" src={ReactIcon} />
                </ImageWrapper>
                <ImageWrapper>
                    SVG
                    <a href="#" onClick={() => console.log("clicked")}>
                        <img width="200" src={Logo} />
                    </a>
                </ImageWrapper>
                <Hello onClick={onClick} text="Click here!" />
                <Button onClick={() => console.log("Hello World")}>Print Hello World</Button>
            </div>
        </AppContainer>
    );
};
