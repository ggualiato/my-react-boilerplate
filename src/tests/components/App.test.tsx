import React from "react";
import { shallow } from "enzyme";
import App from "../../components/App";

test("should render App correctly", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
});
