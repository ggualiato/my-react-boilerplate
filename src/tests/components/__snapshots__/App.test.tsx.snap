// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App correctly 1`] = `ShallowWrapper {}`;
